src/Spark.Engine/Auxiliary/ResourceVisitor.cs

```diff
@@ -76,8 +76,9 @@ private static void scan(object item, string path, Visitor visitor)
             // Do a depth-first traversal of the properties and their contents
             foreach (PropertyInfo property in result)
             {
+                var ienumerableType = property.PropertyType.GetInterface("IEnumerable`1");
                 // If this member is an IEnumerable<Element>, go inside and recurse
-                if (property.PropertyType.GetInterface("IEnumerable`1") != null)
```

Comment:
Strange that this doesn't work already, it seems we should already be filtering properly in `propertyFilter()`. Can you instead of doing another filtering on `IEnumerable<Element>` see how we can improve `propertyFilter`()?